src/vcpkg/base/system.process.cpp

```diff
@@ -677,17 +677,35 @@ namespace vcpkg
             cmd_line,
             wd,
             [&](StringView sv) {
-                auto prev_size = buf.size();
-                Strings::append(buf, sv);
-
-                auto it = std::find(buf.begin() + prev_size, buf.end(), '\n');
-                while (it != buf.end())
+                const auto e = sv.end();
```

Comment:
```suggestion
                const auto last = sv.end();
```

the name `e` confused me; I think `last` is a better name here.